### ISSUE TITLE
Path on Windows 10 does not accept spaces

### ISSUE THREAD

**Issue by IainColledge** [MrYsLab/s2aio#26]
When installing on Windows 10 the client cannot be found and the error states the path is "c:\program".

Installing to C:\ results in a path error of "c:\scratch" as the scratch 2 install path is "c:\Scratch 2".

Renaming this to C:\scratch2\scratch2.exe and setting the path in the configuration file results in the client starting up although it shuts down very fast.

-c no_client allows s2aio to stay up but the blocks are not installed under scratch 2.

Please advise.

**Comment by MrYsLab** [comment]
I just reinstalled both s2aio and the Scratch 2.0 editor on Windows 10 and I am not seeing the error you are reporting.

When installing Scratch from https://scratch.mit.edu/download/scratch2
I accept the defaults:

![Screenshot from 2019-03-16 17-37-09](https://user-images.githubusercontent.com/5189838/54482000-d7e17900-4812-11e9-9d45-88087a5f7bec.png)

I am using pip to install s2aio:

```
pip install s2aio
````

When I start s2aio, I do not see any errors. 

I do not understand your first statement:

> When installing on Windows 10 the client cannot be found and the error states the path is "c:\program".

Could you please explain what steps you are taking to cause this?

**Comment by IainColledge** [comment]
Thanks for the reply.

I removed Scratch 2 and installed as per default.

To keep things simple am not using a virtual env so installing under the system python, version 3.7.2.

I'm also using pip 19.0.3

I install using admin powershell.

When running s2aio I see:

```
C:\Users\Iain>s2aio

pymata_aio Version 2.28 Copyright (c) 2015-2018 Alan Yorinks All rights reserved.

Using COM Port:com17

Initializing Arduino - Please wait...
Arduino Firmware ID: 2.5 FirmataPlus.ino
Auto-discovery complete. Found 22 Digital Pins and 8 Analog Pins


The system cannot find the file c:\program.
```

![2019-03-16](https://user-images.githubusercontent.com/10600605/54482635-a9799300-483e-11e9-8ce6-c4010cffddb9.png)

**Comment by MrYsLab** [comment]
Could you please try installing with a cmd window and not powershell. I suspect that poweshell may be causing the issue. Please let me know if that is successful or not.

**Comment by IainColledge** [comment]
Uninstalled and installed using admin rights with a command shell, exactly the same.

**Comment by MrYsLab** [comment]
I just installed using powershell, powershell with admin, and command with admin and in all cases I do not see the error you are. Here is a screenshot of powershell with admin:
![Screenshot from 2019-03-17 07-58-03](https://user-images.githubusercontent.com/5189838/54490269-c9cf3f00-488a-11e9-92cd-8e1294ae9082.png)

When you installed Python did you check "Add python 3.7 to path"?
![pythonInstall](https://user-images.githubusercontent.com/5189838/54490340-5a0d8400-488b-11e9-8a5d-6d6d581a312b.png)

I also started the program by explicitly stating the path for the relevant files by using the -b option in conjunction with the -c no_client option. After starting the program, I opened up Scratch manually and and manually loaded an s2aio project.

C:\Users\Alan>s2aio -c no_client -b c:\Users\Alan\Appdata\Local\Programs\Python\Python37\Lib\site-packages\s2aio

**Comment by IainColledge** [comment]
I've now installed on a laptop from fresh and works perfectly, looks like a local Python issue.

Thanks for your assistance, will close.

**Comment by jflpimentel** [comment]
I found same issue. There is a problem on Windows, if Python base path has spaces. That is the case when you install Python for all users: the path is "C:\Program Files\Python37".
The solution is to add a double colon before and after self.scratch_executable, on lines 216, 221 and 224 of file __main__.py:
[216] os_string = 'start /b "' + self.scratch_executable + '" ' + self.scratch_project
[221] os_string = 'nohup "' + self.scratch_executable + '" ' + self.scratch_project + ' ' + ' > /dev/null 2>&1 &'
[224] os_string = 'nohup "' + self.scratch_executable + '"> /dev/null 2>&1 &'

**Comment by MrYsLab** [comment]
Thanks for taking the time to track this down. I tried applying your patches, and unfortunately, it resulted in a new set of issues after reinstalling Python for all users. I have updated the python install instructions on the wiki page to click on "Install Now" which installs Python for a single user. I can appreciate why you would want to install for all users, but at this point, I can only support install for a single user.

Thanks again.
